ComponentKit/Core/Scope/CKComponentScope.mm

```diff
@@ -17,6 +17,7 @@
 CKComponentScope::~CKComponentScope()
 {
   if (_threadLocalScope != nullptr) {
+    [_scopeHandle resolve];
```

Comment:
This is clever! Cool approach.

So we completely avoid the problem of a partially initialized component by deferring "resolution" until the component scope is destroyed (and the component has finished initializing). Neat.